### DISCUSSION TITLE
I'd like to ask you a question about how to use CPLEX.

### DISCUSSION THREAD

**Discussion by 123bjmy** [yalmip/YALMIP#965] [General discussion] [answered]
I'd like to ask you a question about how to use CPLEX. The optimization model is as follows. The objective function contains quadratic terms, and there are a few nonlinear constraints in the constraints. How can we use CPLEX optimization.

**Comment by johanlofberg** [comment, marked as answer]
Cplex can solve quadratic (or SOCP-representable to be precise) problems, hence not general nonlinear models.

Using cplex is no different from using any other solver in YALMIP
https://yalmip.github.io/tutorial/basics/

**Comment by 123bjmy** [comment]
There are several nonlinear constraints in the constraints. The objective function is in square form, and CPLEX can be solved directly?


| |
bjmy123456
|
|
***@***.***
|
签名由网易邮箱大师定制
On 8/9/2021 01:54，Johan ***@***.***> wrote：

Cplex can solve quadratic (or SOCP-representable to be precise) problems, hence not general nonlinear models.

Using cplex is no different from using any other solver in YALMIP
https://yalmip.github.io/tutorial/basics/

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by johanlofberg** [reply]
As I said, it requires that the constraints are (mixed-integer) SOCP-representable

**Comment by 123bjmy** [comment]
Because the independent variables of the problem are all continuous, the constraints are not mixed-integer
| |
bjmy123456
|
|
***@***.***
|
签名由网易邮箱大师定制
On 8/9/2021 02:00，Johan ***@***.***> wrote：

As I said, it requires that the constraints are (mixed-integer) SOCP-representable

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by johanlofberg** [reply]
i have no idea what that comment is about. As I said, cplex can only solve MISOCP/nonconvex QP and nothing else

**Comment by 123bjmy** [comment]
There are only continous variables in the optimization problem. Can't you use CPLEX to solve it


| |
bjmy123456
|
|
***@***.***
|
签名由网易邮箱大师定制
On 8/9/2021 02:07，Johan ***@***.***> wrote：

i have no idea what that comment is about. As I said, cplex can only solve MISOCP/nonconvex QP and nothing else

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by johanlofberg** [reply]
I have already told you, Cplex can only handle SOCP models (and nonconvex quadratic objective) and nothing else. Hence, if your nonlinearities are not SOCP-representable (or more generally MISOCP-representable) you cannot use Cplex
